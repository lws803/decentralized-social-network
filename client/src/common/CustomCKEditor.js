import React from "react";

import CKEditor from "@ckeditor/ckeditor5-react";
import ClassicEditor from "ckeditor5-for-code-and-image/build/ckeditor";

import { ArticleBox } from "./CommonStyles";
import {ChangeFontForAllHeadings} from "./CKPlugins";

class CustomCKEditor extends React.Component {
  render() {
    return (
      <ArticleBox>
        <CKEditor
          editor={ClassicEditor}
          onInit={editor => {
            console.log("Editor is ready to use!", editor);
          }}
          config={{
            extraPlugins: [ChangeFontForAllHeadings],
            removePlugins: ["Title"],
            toolbar: {
              items: [
                "heading",
                "|",
                "fontSize",
                "bold",
                "italic",
                "link",
                "bulletedList",
                "numberedList",
                "underline",
                "blockQuote",
                "undo",
                "redo",
                "|",
                "indent",
                "outdent",
                "|",
                "imageUpload",
                "code",
                "codeBlock",
                "mediaEmbed",
              ],
            },
            language: "en",
            image: {
              toolbar: [
                "imageTextAlternative",
                "imageStyle:full",
                "imageStyle:side",
              ],
            },
            simpleUpload: {
              uploadUrl: process.env.REACT_APP_API_URL + "/image_upload",
              withCredentials: false,
            },
          }}
          {...this.props}
        />
      </ArticleBox>
    );
  }
}

export default CustomCKEditor;
