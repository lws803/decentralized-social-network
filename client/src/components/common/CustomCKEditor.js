import React from "react";

import CKEditor from "@ckeditor/ckeditor5-react";
import ClassicEditor from "ckeditor5-for-code-and-image/build/ckeditor";

import { ArticleBox } from "./CommonStyles";
import { ChangeFontForAllHeadings } from "./CKPlugins";

const CustomCKEditor = props => {
  return (
    <ArticleBox>
      <CKEditor
        editor={ClassicEditor}
        onInit={editor => {
          console.log("Editor is ready to use!", editor);
        }}
        config={{
          extraPlugins: [ChangeFontForAllHeadings],
          removePlugins: ["Title"],
          toolbar: {
            items: [
              "heading",
              "|",
              "fontSize",
              "bold",
              "italic",
              "link",
              "bulletedList",
              "numberedList",
              "underline",
              "blockQuote",
              "undo",
              "redo",
              "|",
              "indent",
              "outdent",
              "|",
              "imageUpload",
              "code",
              "codeBlock",
              "mediaEmbed",
            ],
          },
          language: "en",
          image: {
            toolbar: [
              "imageTextAlternative",
              "imageStyle:full",
              "imageStyle:side",
            ],
          },
          simpleUpload: {
            uploadUrl: sessionStorage.getItem("currentAPI") + "/image_upload",
            withCredentials: false,
          },
        }}
        {...props}
      />
    </ArticleBox>
  );
};

export default CustomCKEditor;
