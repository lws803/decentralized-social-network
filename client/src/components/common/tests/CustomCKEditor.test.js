import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";

import CustomCKEditor from "../CustomCKEditor";

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Custom CK Editor component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<CustomCKEditor />);
    expect(wrapper).toMatchSnapshot();
  });

  it("accepts inputs correctly", () => {
    const onDataChange = jest.fn();
    const wrapper = shallow(<CustomCKEditor onChange={onDataChange} data="" />);
    const event = {
      preventDefault() {},
      target: { value: "value" },
    };
    wrapper.find("a").simulate("change", event);
    expect(onDataChange.mock.calls.length).toEqual(1);
  });
});
