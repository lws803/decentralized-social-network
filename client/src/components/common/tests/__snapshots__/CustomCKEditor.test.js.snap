// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom CK Editor component renders correctly 1`] = `
<styled.div>
  <a
    config={
      Object {
        "extraPlugins": Array [
          [Function],
        ],
        "image": Object {
          "toolbar": Array [
            "imageTextAlternative",
            "imageStyle:full",
            "imageStyle:side",
          ],
        },
        "language": "en",
        "removePlugins": Array [
          "Title",
        ],
        "simpleUpload": Object {
          "uploadUrl": "null/image_upload",
          "withCredentials": false,
        },
        "toolbar": Object {
          "items": Array [
            "heading",
            "|",
            "fontSize",
            "bold",
            "italic",
            "link",
            "bulletedList",
            "numberedList",
            "underline",
            "blockQuote",
            "undo",
            "redo",
            "|",
            "indent",
            "outdent",
            "|",
            "imageUpload",
            "code",
            "codeBlock",
            "mediaEmbed",
          ],
        },
      }
    }
    editor={[Function]}
    onInit={[Function]}
  />
</styled.div>
`;
